map.h

```diff
@@ -2,94 +2,52 @@
 #include "splashkit.h"
 #include <fstream>
 #include <iostream>
+#include <memory>
+#include "block.h"
 using namespace std;
-// Define screen size and tilesize
-#define SCREEN_WIDTH  1600 // 50 tiles
-#define SCREEN_HEIGHT  896 // 28 tiles
-#define TILESIZE        16 // pixels
-#define MAP_WIDTH       40 // tiles
-#define MAP_HEIGHT      32 // tiles
 
 /**
  * @brief Map class
  * 
  */
-class Map
+class Level_Ojects_Map
 {
-    private:
-        // stores the background image
-        bitmap background;
-        // 2D array of level design from text file
-        vector<vector<int> > map_array;
+    protected:
+        vector<vector<int> > map_array; 
+        int screen_width = 1600;
+        int screen_height = 896;
+        int tile_size = 64;
+        int map_width;
+        int map_height;
+        string type;
 
     public:
         /**
          * @brief Default Constructor
          */
-        Map(){}
+        Level_Ojects_Map(){};
 
         /**
          * @brief Destructor
          */
-        ~Map(){}
+        ~Level_Ojects_Map()
+        {
+            map_array.clear();
+        };
 
         /**
          * @brief First Overloaded Constructor
          * @param level        The text file used to define the level
          * @param sprite_sheet The sprite sheet used to build the map
          * 
          */
-        Map(string level, string background)
+        Level_Ojects_Map(string level, int tile_size)
         {
-            // use new_level function to load the map to new_game.map_array
+            this->tile_size = tile_size;
+            this->map_width = (screen_width / tile_size);
+            this->map_height = (screen_height / tile_size);
```

Comment:
This is fine when the entirety of the map is displayed inside the window, but what happens should the camera pan?